integrated-benchmarks/README.md

```diff
@@ -0,0 +1,62 @@
+# OpenTelemetry Integrated Benchmarks
+
+Related:
+
+- https://github.com/open-telemetry/opentelemetry-js/issues/4741
+
+Integrated benchmarks are a set of benchmarks that run simple Node.js web servers with and without OpenTelemetry instrumentation. They use a combination of [Crystal](https://crystal-lang.org/), [Nix](https://nixos.org/), [bombardier](https://github.com/codesenberg/bombardier) to spawn Node.js servers using frameworks like HTTP and Express with and without basic OpenTelemetry usage.
+
+For each commit the benchmarks will run in CI, on pull requests the results are outputted to the console, and on main they are committed into the `benchmarks.md` file.
+
+## Why Integrated Benchmarks?
+
+OpenTelemetry JS has a set of benchmarks located here https://open-telemetry.github.io/opentelemetry-js/benchmarks/, however, they don't give a good reflection of how processing, exporting, and creating spans in a web server affect the performance, and more benchmark indavidual components of OpenTelemetry.
+
+These Integrated benchmarks aim to give a better reflection of how OpenTelemetry affects the performance of a web server by providing a simple web server with and without OpenTelemetry instrumentation, showing the performance difference between the two.
+
+With these benchmarks users can see how OpenTelemetry affects the performance of a web server, and how different components of OpenTelemetry affect the performance of a web server.
+
+## How to run benchmarks
+
+To run the benchmarks locally, you will need to have [Nix](https://nixos.org/) installed. Once you have Nix installed, you can run the following command to run the benchmarks:
+
+- https://nixos.org/download/
+
+```sh
+cd ./integrated-benchmarks && nix-shell --quiet --run ./run.cr
+```
+
+> Its best to use the pipeline to run the benchmarks, as it will ensure that the benchmarks are run in the same environment as CI. Running locally may give different results and consume all your resources.
```

Comment:
The committed initial benchmarks in `benchmarks.md` are from running locally. Ideally, we should let the main commit in the benchmarks and use the main machine as the source of truth. On merge the benchmarks file would be updated and should reflect more reliable data.